src/Server/TCPHandler.cpp

```diff
@@ -796,6 +799,8 @@ void TCPHandler::processOrdinaryQueryWithProcessors()
             }
         }
 
+        std::lock_guard lock(task_callback_mutex);
```

Comment:
Create unique_lock outside the scope and call `lock` method on it here so you don't release the lock and take it instantly again.